Dockerfile

```diff
@@ -10,14 +10,28 @@ WORKDIR /usr/src
 ## Setup Home Assistant Core dependencies
 COPY requirements.txt homeassistant/
 COPY homeassistant/package_constraints.txt homeassistant/homeassistant/
+# hadolint ignore=DL4006
+RUN \ 
+   grep -v '^[#-]' < homeassistant/requirements.txt | xargs -P 6 -n 4 -- \
+   pip3 download --no-index --only-binary=:all: \
+   --find-links "${WHEELS_LINKS}" --use-deprecated=legacy-resolver \
+   --cache-dir "/tmp/pip_cache" ; exit 0
 RUN \
-    pip3 install --no-cache-dir --no-index --only-binary=:all: --find-links "${WHEELS_LINKS}" \
+    pip3 install --cache-dir "/tmp/pip_cache" --no-index --only-binary=:all: --find-links "${WHEELS_LINKS}" \
     -r homeassistant/requirements.txt --use-deprecated=legacy-resolver
 COPY requirements_all.txt homeassistant/
+# hadolint ignore=DL4006
+RUN \ 
+   grep -v '^[#-]' < homeassistant/requirements_all.txt | xargs -P 8 -n 120 -- \
+   pip3 download --no-index --only-binary=:all: \
+   --find-links "${WHEELS_LINKS}" --use-deprecated=legacy-resolver \
+   --cache-dir "/tmp/pip_cache" ; exit 0
 RUN \
-    pip3 install --no-cache-dir --no-index --only-binary=:all: --find-links "${WHEELS_LINKS}" \
+    pip3 install --cache-dir "/tmp/pip_cache" --no-index --only-binary=:all: --find-links "${WHEELS_LINKS}" \
     -r homeassistant/requirements_all.txt --use-deprecated=legacy-resolver
 
+RUN rm -rf /tmp/pip_cache
```

Comment:
So one feedback I got from @jesserockz is that since it's different run steps, the cache is still stored in the previous layers. So it means it's still there.